src/integrationTest/java/org/opensearch/security/SearchOperationTest.java

```diff
@@ -999,8 +1003,8 @@ public void shouldIndexDocumentInBulkRequest_partiallyPositive() throws IOExcept
         auditLogsRule.assertExactlyOne(userAuthenticated(LIMITED_WRITE_USER).withRestRequest(POST, "/_bulk"));
         auditLogsRule.assertExactlyOne(grantedPrivilege(LIMITED_WRITE_USER, "BulkRequest"));
         auditLogsRule.assertExactly(2, grantedPrivilege(LIMITED_WRITE_USER, "CreateIndexRequest"));
-        auditLogsRule.assertExactly(6, auditPredicate(INDEX_EVENT).withEffectiveUser(LIMITED_WRITE_USER));
```

Comment:
To me its a bug if we don't have a consistent count.  If there is some kind of retry or reissuing or other dynamic contribution to the test, we should be able to measure it and verify that its behaving consistently with these assertions.

For example we might want to change `waitForSnapshotCreation` so it returns the number of calls issued, this might be causing the inconsistency, then we can use that value to indicate how many events are in the audit log for those specific scenarios.

Reply:
+1. Is there a way we can make this more concrete?

Reply:
Seems like issue in this test was caused by transport requests between the cluster nodes. Forcing rest client to communicate through one of the managers node helps to stabilize audit logs count. (cluster.setPrimaryNode() is used for that) Other helpful thing was setting up index to use less shards.